src/commands/create.js

```diff
@@ -76,153 +48,121 @@ export default asyncCommand({
 	},
 
 	async handler(argv) {
-		let template = TEMPLATES[argv.type];
-
-		if (!template) {
-			throw Error(`Unknown app template "${argv.type}".`);
-		}
-
-		let target = path.resolve(process.cwd(), argv.dest || argv.name);
+		let isYarn = argv.yarn && hasCommand('yarn');
+		let cwd = argv.cwd ? resolve(argv.cwd) : process.cwd();
+		let target = argv.dest && resolve(cwd, argv.dest);
+		let exists = target && isDir(target);
+
+		if (target) {
+			if (exists && !argv.force) {
+				return error('Refusing to overwrite current directory! Please specify a different destination or use the `--force` flag', 1);
+			}
 
-		let exists = false;
-		try {
-			exists = (await fs.stat(target)).isDirectory();
-		}
-		catch (err) {}
-
-		if (exists && argv.force) {
-			const question = {
-				type: 'confirm',
-				name: 'enableForce',
-				message: `You are using '--force'. Do you wish to continue?`,
-				default: false,
-			};
-
-			let { enableForce } = await inquirer.prompt(question);
-
-			if (enableForce) {
-				process.stdout.write('Initializing project in the current directory...\n');
-			} else {
-				process.stderr.write(chalk.red('Error: Cannot initialize in the current directory\n'));
-				process.exit(1);
+			if (exists && argv.force) {
+				let { enableForce } = await prompt({
+					type: 'confirm',
+					name: 'enableForce',
+					message: `You are using '--force'. Do you wish to continue?`,
+					default: false
+				});
+
+				if (enableForce) {
+					process.stdout.write('Initializing project in the current directory...\n');
+				} else {
+					return error('Refusing to overwrite current directory!', 1);
+				}
 			}
+		} else {
+			// TODO: interactive
```

Comment:
@developit Should we really do this? This make it more complex for us to handle.

Reply:
Let's save this discussion for a future PR. I want to break it up instead of making this (even more) massive.